src/Application.php

```diff
@@ -213,11 +213,11 @@ public function routeMiddleware(ServerRequestInterface $request, ResponseInterfa
      * @param callable|string $middleware Middleware (or middleware service name) to associate with route.
      * @param null|array $methods HTTP method to accept; null indicates any.
      * @return Router\Route
-     * @throws InvalidArgumentException if $path is not a Router\Route AND middleware is null.
+     * @throws InvalidArgumentException if $path is not a Router\Route|string AND middleware is null.
      */
     public function route($path, $middleware = null, array $methods = null)
     {
-        if (! $path instanceof Router\Route && null === $middleware) {
+        if ((! is_string($path) || ! $path instanceof Router\Route) && null === $middleware) {
             throw new InvalidArgumentException(sprintf(
```

Comment:
yes, but the InvalidArgumentException itself detected as never imported in travis after the change. while the change not valid, the InvalidArgumentException usage is need to be fixed.